### PR TITLE
Fix use of deprecated header attributes in nevow.compression.

### LINKED ISSUE #74: Add some extra compatibility assistance in FakeRequest

### LINKED ISSUE #75: nevow.compression refers to deprecated header attributes

### PR DESCRIPTION
**Pull request by mithrandi**
Fixes #74, fixes #75.

Also add better compatibility support to FakeRequest for use of the old and new header attributes, and fix test code that tripped over the deprecation warnings.

### CODE REVIEW

File: nevow/testutil.py [hunk 1]
```diff
@@ -246,6 +248,46 @@ def isSecure(self):
...
+    @property
+    def received_headers(self):
+        """
+        Transform the L{Headers}-style C{requestHeaders} attribute into a
+        deprecated C{dict}-style C{received_headers} attribute.
+        """
+        self._warnHeaders("received_headers", "requestHeaders")
+        return appserver._DictHeaders(self.requestHeaders)
+
+
```
Comment (jonathanj) [inline nevow/testutil.py]: You're missing some whitespace here.

Comment (mithrandi) [inline nevow/testutil.py]: Fixed.

**Comment by jonathanj** [comment]
This seems fine to me. Does `nevow.compression` give you anything you can't get by running your application behind nginx or similar? Perhaps deprecating it is not the worst idea, Nevow could certainly do with less code to maintain.

(Also I'm sure there is a joke in here about being doomed to maintain the code you wrote for the rest of your life.)

**Comment by mithrandi** [comment]
Well, it gets you "not having to run a web server made out of a huge pile of C code". You're probably better off using `twisted.web` support these days, though; but I guess if you have a Nevow application, you need to use `nevow.compression`, I don't think the interfaces are compatible enough to mix and match.

The code is in fairly good shape, though, so it's not really a drag to maintain compared with other stuff.

**Pull request merged**